pkg/flag/options.go

```diff
@@ -249,6 +250,24 @@ func getValue(flag *Flag) any {
 	return viper.Get(flag.ConfigName)
 }
 
+func explodeGlob(paths []string, dir string) []string {
+	var exploded []string
+
+	for _, path := range paths {
+		if !strings.Contains(path, "*") {
+			exploded = append(exploded, path)
+			continue
+		}
+		if globPaths, err := filepath.Glob(filepath.Join(dir, path)); err == nil {
```

Comment:
This approach is interesting, but I think it causes an issue with container images. Please correct me if I'm wrong.

What if using filepath.Match here?
https://github.com/aquasecurity/trivy/blob/b0cebec324d96a9699ad5262020fcb4e55e4e459/pkg/fanal/walker/walk.go#L56

```
for _, pattern := range w.skipFiles {
	if filepath.Match(pattern, filePath) {
		return true
	}
}
return false
```

Reply:
Yeah that's a good idea. I thought about implementing it in the analyzer (fanal) first too. `filepath.Match` works too. I implemented that here: https://github.com/aquasecurity/trivy/pull/3866/commits/09ba3285b1c340dce4124a4d58cf2cba7bd93d3a